libsolidity/parsing/Parser.cpp

```diff
@@ -102,6 +123,7 @@ ASTPointer<SourceUnit> Parser::parse(shared_ptr<Scanner> const& _scanner)
 
 ASTPointer<PragmaDirective> Parser::parsePragmaDirective()
 {
+	RecursionGuard recursionGuard(*this);
```

Comment:
Hm, just realised some of these are actually not that useful, pragma, import, etc.